### PR TITLE
Documentation with Dokka and KDoc

### PR DESCRIPTION
**Pull request by BasBuijsen**
This adds the setup of Dokka. This generates HTML based on the KDoc comments provided for every function and class. For now i didn't add them yet to all the classes since that is a huge chunk of work. I will incrementally add documentation, but for now only showing the signatures should suffice.

The publishing of the generated HTML still needs to be setup. I need @nbransby for that so that is out of my control.

Since all publicly available classes and functions will be shown in the documentation, [this change](https://github.com/GitLiveApp/firebase-kotlin-sdk/issues/535) will also improve the documentation a lot. It will make sure that only the classes and functions that are actually exposed will be documented.

### CODE REVIEW

**Comment by nbransby** [comment]
@Reedyuk what do you think is the best way to publish the documentation? is there a GitHub Action we can use to publish the dokka output to github pages? 

@BasBuijsen do you have experience with publishing to GitHub pages?

**Comment by nbransby** [comment]
@BasBuijsen build failed due to the function you made internal that are needed by inline functions (can use [@PublishedApi](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-published-api/) here)

**Comment by BasBuijsen** [comment]
@nbransby i dont have experience with publishing to github pages but from what ive seen it is very easy. There is a ready to go action that will do all the magic if you set the environment variables correctly, so i think that would be the easiest and probably cheapest solution (there is one website free per gh organisation or user).

I will fix the JS build this afternoon, after work :)

**Comment by Reedyuk** [comment]
https://github.com/kwebio/kweb-core/blob/master/.github/workflows/build.yml#L25

**Comment by BasBuijsen** [comment]
@Reedyuk thanks for adding the deployment steps!

Ive added  documentation for nearly all modules now. Only authentication, analytics and firestore are left to do.

**Comment by Reedyuk** [comment]
I think rebase didnt work correctly, as its not compiling on the github servers.

File: firebase-analytics/documentation.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Module firebase-analytics
```
Comment (nbransby) [inline firebase-analytics/documentation.md:1]: can we use the official documentation module introductions here? they should be found in the package-info.java files in the android sdk

Comment (BasBuijsen) [reply firebase-analytics/documentation.md:1]: Yes sure! I will change that :)

**Comment by BasBuijsen** [comment]
> I think rebase didnt work correctly, as its not compiling on the github servers.

Yes i made a mistake while resolving the merge conflict. Will fix it :)

**Comment by Reedyuk** [comment]
> > I think rebase didnt work correctly, as its not compiling on the github servers.
> 
> Yes i made a mistake while resolving the merge conflict. Will fix it :)

get this building on the ci and i will merge in and make sure the docs deploy correctly.

File: firebase-analytics/documentation.md (hunk 1 above)
Comment (BasBuijsen) [reply firebase-analytics/documentation.md:1]: The package-info.java files in the android sdk's don't seem to contain the module introductions that you mention. Where did you find that?

**Comment by Reedyuk** [comment]
@BasBuijsen - can you rebase this pull request please

File: firebase-analytics/documentation.md (hunk 1 above)
Comment (Reedyuk) [reply firebase-analytics/documentation.md:1]: should we take the ones from the android sdk?
https://github.com/firebase/firebase-android-sdk/tree/main/docs/ktx

Comment (BasBuijsen) [reply firebase-analytics/documentation.md:1]: Thats only documentation for the kotlin extensions so i think thats not quite right

Comment (nbransby) [reply firebase-analytics/documentation.md:1]: I assumed they would have intros in the packages, but they don't they are just empty - you can see here https://firebase.google.com/docs/reference/android/com/google/firebase/firestore/package-summary

We can just leave ours empty too in this case

Comment (BasBuijsen) [reply firebase-analytics/documentation.md:1]: okay! ill leave it like this then

**Comment by BasBuijsen** [comment]
@nbransby or @Reedyuk would it be okay to merge this pr now?

I will add comments to Auth and Firestore next week. Analytics is difficult because it doesn't have an open source codebase so i cant really copy the comments.

**Review by Reedyuk — APPROVED** [review]

**Comment by BasBuijsen** [comment]
@Reedyuk when its deployed can you send me the url so i can add it to the readme when i create a new pr with the final documentation changes next week?

**Pull request merged**